src/store/actions.js

```diff
@@ -707,4 +708,9 @@ export default {
 		commit('removeMailbox', { id: mailbox.databaseId })
 		commit('addMailbox', { account, mailbox: newMailbox })
 	},
+	async moveEnvelope({ dispatch }, { envelope, destMailboxId }) {
+		await moveMessage(envelope.databaseId, destMailboxId)
+		dispatch('syncEnvelopes', { mailboxId: envelope.mailboxId })
```

Comment:
IMO the code that dispatches a `moveEnvelope` should subsequently trigger the sync, not the action itself

Reply:
I agree. Ideally, only the affected envelopes should be updated but I couldn't get it to work reliably. Synching the whole mailbox maybe a bit overkill.